src/status_im2/contexts/chat/messages/navigation/view.cljs

```diff
@@ -14,42 +14,44 @@
-                                   (rf/sub [:chats/photo-path chat-id]))
-        opacity-animation        (reanimated/interpolate scroll-y
-                                                         [style/navigation-bar-height
-                                                          (+ style/navigation-bar-height 30)]
-                                                         [0 1]
-                                                         {:extrapolateLeft  "clamp"
-                                                          :extrapolateRight "extend"})
-        banner-opacity-animation (reanimated/interpolate scroll-y
-                                                         [(+ style/navigation-bar-height 150)
-                                                          (+ style/navigation-bar-height 200)]
-                                                         [0 1]
-                                                         {:extrapolateLeft  "clamp"
-                                                          :extrapolateRight "extend"})
-        translate-animation      (reanimated/interpolate scroll-y
-                                                         [(+ style/navigation-bar-height 25)
-                                                          (+ style/navigation-bar-height 100)]
-                                                         [50 0]
-                                                         {:extrapolateLeft  "clamp"
-                                                          :extrapolateRight "clamp"})
-        title-opacity-animation  (reanimated/interpolate scroll-y
-                                                         [0 50]
-                                                         [0 1]
-                                                         {:extrapolateLeft  "clamp"
-                                                          :extrapolateRight "clamp"})]
+         :as   chat}              (rf/sub [:chats/current-chat-chat-view])
+        shell-animation-complete? (rf/sub [:shell/animation-complete? (:chat-type chat)])
+        all-loaded?               (when shell-animation-complete?
+                                    (rf/sub [:chats/all-loaded? (:chat-id chat)]))
+        display-name              (if (= chat-type constants/one-to-one-chat-type)
+                                    (first (rf/sub [:contacts/contact-two-names-by-identity chat-id]))
+                                    (str emoji " " chat-name))
+        online?                   (rf/sub [:visibility-status-updates/online? chat-id])
+        contact                   (when-not group-chat
+                                    (rf/sub [:contacts/contact-by-address chat-id]))
+        photo-path                (when-not (empty? (:images contact))
+                                    (rf/sub [:chats/photo-path chat-id]))
+        opacity-animation         (reanimated/interpolate scroll-y
```

Comment:
I find this whole block hard to read, perhaps you could pull these animations into helper functions?

Reply:
could you elaborate? 
all those animations are pretty different and use different values, and I think moving them to separate functions and hiding them somewhere else would make it harder to understand what's going on

Reply:
actually I think it's just from the smaller window in the git diff view it appears this way, looking at the file in isolation it's fine to read. So I think it's fine to leave as is 👌